### PR TITLE
Save file before starting debug session

### PR DESCRIPTION
**Pull request by tgodzik**
Fixes https://github.com/scalameta/metals-vscode/issues/524

### CODE REVIEW

**Comment by tgodzik** [comment]
@ckipp01 how is this handled in the vim DAP extension?

File: src/scalaDebugger.ts [hunk 1]
```diff
@@ -31,10 +31,13 @@ export async function start(
   noDebug: Boolean,
   ...parameters: any[]
 ): Promise<Boolean> {
-  return vscode.commands
-    .executeCommand<DebugSession>(
-      ServerCommands.DebugAdapterStart,
-      ...parameters
+  return commands
+    .executeCommand("workbench.action.files.save")
```
Comment (gabro) [inline src/scalaDebugger.ts:35]: I wonder if we should do this unconditionally, or prompt the user and offer to save.

Comment (tgodzik) [reply src/scalaDebugger.ts:35]: So it seems that vscode does it automatically anyways when the debug session is run, which means that even with prompting the user the end result will be the same.

Comment (gabro) [reply src/scalaDebugger.ts:35]: 👍 alright then

**Review by gabro — APPROVED** [review]

**Comment by ckipp01** [comment]
> @ckipp01 how is this handled in the vim DAP extension?

I actually haven't checked before, but it seems that if you have unsaved changes in the buffer, it just runs with the last known state. However, that may be quite tricky to change there. Mainly because `nvim-metals` isn't in charge of the actual running at all, just the setup. It basically hands over a couple functions as configuration to `nvim-dap` and then the user triggers the entire thing via `nvim-dap`, not `nvim-metals`. So I'll need to actually poke around `nvim-dap` a bit to see if maybe that could be handled there.

**Comment by tgodzik** [comment]
> > @ckipp01 how is this handled in the vim DAP extension?
> 
> I actually haven't checked before this, but it seems that if you have unsaved changes in the buffer, it just runs with the last known state. However, that may be quite tricky to change there. Mainly because `nvim-metals` isn't in charge of the actual running at all, just the setup. It basically hands over a couple functions as configuration to `nvim-dap` and then the user triggers the entire thing via `nvim-dap`, not `nvim-metals`. So I'll need to actually poke around `nvim-dap` a bit to see if maybe that could be handled there.

I think this is perfectly fine. The main issue would be if it automatically saved the buffers like VS Code, but otherwise if it's totally in the users control then I then we shouldn't change anything.

**Pull request merged**